### ISSUE TITLE
Using a local data provider, a file collection, and a container, symbolic links break as the container doesn't know where the link goes

### ISSUE THREAD

**Issue by shots47s** [aces/cbrain#692]
When a local Data Provider has a task launched on the same machine, in the Bourreau work directory, a symbolic link is created pointing to the data, so that nothing needs to be moved.  The issue with this is that unless the directory where the original data is mounted, this link is pointing to nothing once inside the container and does not represent the directory anymore as it is a broken link.

One possible solution is to also mount the DataProvider directory in the container, which will make the link active again.

**Comment by prioux** [comment]
This is supposed to work properly, we made sure to mount all needed paths inside the containers.

Can you provide the following information:

1) what container system? docker or singularity?
2) full path to the Bourreau's DP cache dir
3) full path to the local data provider dir
4) full path to the Bourreau's task workdir root (aka gridshare)
5) full path to the task

**Comment by prioux** [comment]
I think this might be a duplicate of #542

**Comment by shots47s** [comment]
This is running all in one vagrant image, with the BrainPortal, Data Provider, and Bourreau running on the same machine.

1. Docker
2. /home/ubuntu/BorCache
3. /home/ubuntu/DPDir
4. /home/ubuntu/BorWork
5. Not sure what you mean by this


-- 
Shawn T. Brown, PhD.
Chief Software Architect / CBRAIN
McGill Centre for Integrative Neuroscience (MCIN)
McGill University
Montreal, QC, Canada
Email: stbrown@mcin.ca
Phone: 514-237-2960

**Comment by prioux** [comment]
Ok; for the moment there's a simple workaround: change the type of the data provider for the network type, instead of the local (or smart) type. This need to be one with the console, but it's simple:

```ruby
d = DataProvider.find(123) # change 123 for the proper ID
d.type = 'EnCbrainSshDataProvider'  # adjust; the network type must match the original type
d.save
```

The DP types are shown in the LEFT side of this diagram:

https://github.com/aces/cbrain/wiki/Data-Providers#type_dp

**Comment by prioux** [comment]
You'll have to relaunch your task too.

**Comment by shots47s** [comment]
Thanks trying now.

**Comment by prioux** [comment]
Fixed by #707

**Comment by prioux** [comment]
Oops, not merged yet.

**Comment by prioux** [comment]
Fixed by #707
